class.c

```diff
@@ -915,6 +915,7 @@ rb_include_class_new(VALUE module, VALUE super)
     if (BUILTIN_TYPE(module) == T_ICLASS) {
 	module = RBASIC(module)->klass;
     }
+    assert(!RB_TYPE_P(module, T_ICLASS));
```

Comment:
could you use `RUBY_ASSERT()` which is disabled with normal build?
or do you want to enable it always?